src/utils/Firebase/firebase.js

```diff
@@ -36,23 +31,83 @@ class Firebase {
 
   doSignOut = () => this.auth.signOut();
 
-  // EXAMPLE ENDPOINT
-  getLesson = () => this.db.collection("custom_lesson");
-
-  //Gets last mastered lesson of a given student.
-  getLastMastered = deploymentAccountId =>
+  //Gets the most recent lesson that a student has completed
+  getLastMasteredLesson = deploymentAccountId =>
     this.db
       .doc(`deployment_account/${deploymentAccountId}/`)
       .get()
-      .then(documentSnapshot => {
-        documentSnapshot.get("profile.glenLearn.lastMasteredLesson");
-      });
+      .then(deploymentDoc => {
+        const lastMasteredLesson = deploymentDoc.get(
+          "profile.glenLearn.lastMasteredLesson"
+        );
+        return new Promise((resolve, reject) => {
+          if (deploymentDoc) {
+            resolve(lastMasteredLesson);
+          } else {
+            reject(Error("Deployment doc does not exist"));
+          }
+        });
+      })
+      .catch(error => console.log("Error getting student account: ", error));
```

Comment:
something that makes me unsure about the way you're writing endpoint is that Jason doesn't format his endpoints this way and we don't want to deviate too much from the way they do things